src/middleware/authorization.js

```diff
@@ -0,0 +1,24 @@
+import { isDevelopment } from '../config';
+import { handleRequest, FALLBACK_API_KEY, findDeveloper } from '../controllers/developers/utils';
+
+export default async (req, res, next) => {
+  try {
+    let { apiToken: apiKey } = handleRequest(req);
+
+    if (!apiKey && isDevelopment) {
+      apiKey = FALLBACK_API_KEY;
+    }
+
+    // Check if API key belongs to the requested developer
+    const developer = await findDeveloper(apiKey);
+
+    if (developer.length < 1) {
+      throw new Error('Invalid API Key. Check your API Key and try again', 403);
+    }
+    console.info('Developer found', developer);
+    req.developer = developer;
+    return next();
+  } catch (err) {
+return res.status(400).send({ error: err.message });
```

Comment:
it seems like you're not using the status codes when you catch the errors that include a status code - you would want to do something like this

```suggestion
let status = 400;
if ('cause' in err) {
  status = cause;
}
return res.status(status).send({ error: err.message });
```